### PR TITLE
[JS] Topic: Added topics Iterables, Sets and Maps

### LINKED ISSUE #1833: [JS]Topic: Iterables, Sets and Maps in JavaScript Objects
### Description

 # Iterables, Sets and Maps in JavaScript Objects
In this, We'll discuss Iterables, Sets, and Maps in JavaScript Objects

## Directory
Topics/5.Syntax/Objects/

# Description
I want to update the already made README file for Javascript  objects with some additional contents

### Domain

JavaScript

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by daminimehra**
<hr>

## Description 📜

In this, I have discuss Iterables, Sets, and Maps in JavaScript Objects
Fixes #1833 Issue

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Doc#umentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [ ] C/#CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [x] Java#Script
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by daminimehra** [comment]
@paramsiddharth sir pls review it!

**Comment by paramsiddharth** [comment]
@Creatoon @dakshsethi @s4-2 Please review.

File: JavaScript/Topics/Data_Structures/README.md [hunk 1]
```diff
@@ -0,0 +1,289 @@
+
```
Comment (paramsiddharth) [inline JavaScript/Topics/Data_Structures/README.md:1]: Please add a title.
```suggestion
# Data Structures in JavaScript

```

**Review by paramsiddharth — CHANGES_REQUESTED** [review]

**Review by s4-2 — APPROVED** [review]
Looks good to me:)

**Review by paramsiddharth — APPROVED** [review]

**Comment by daminimehra** [comment]
I have modified this README.md file again, @paramsiddharth sir with some new content. Hope you like it!

**Comment by paramsiddharth** [comment]
@Creatoon @dakshsethi Please review.

**Review by dakshsethi — APPROVED** [review]
Looks good to me!! 🔥🔥

File: JavaScript/Topics/Data_Structures/README.md (hunk 1 above)
Comment (Creatoon) [reply JavaScript/Topics/Data_Structures/README.md:1]:
```suggestion
```
remove 1st empty line

File: JavaScript/Topics/Data_Structures/README.md [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+If an object isn’t technically an array, but represents a collection (list, set) of something, then ``for..of`` is a great syntax to loop over it, so let’s see how to make it work.
+
+## **Iterating Over a String**
+You can use a ``for..of`` loop to iterate over the elements of a string:
+
+**For Example**
+```js
+for (const x of "GWOC") {
+  // code block to be executed
```
Comment (Creatoon) [inline JavaScript/Topics/Data_Structures/README.md:18]:
```suggestion
  // code block to be executed
  console.log(x);
```

File: JavaScript/Topics/Data_Structures/README.md [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+}
```
Comment (Creatoon) [inline JavaScript/Topics/Data_Structures/README.md:19]:
```suggestion
}
```
Also write output for this, just for this example, rest is understandable

File: JavaScript/Topics/Data_Structures/README.md [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+```
+
+## **Iterating Over an Array**
+You can use a **for..of** loop to iterate over the elements of an Array:
+
+**For Example**
+```js
+for (const x of [1,2,3,4,5] {
```
Comment (Creatoon) [inline JavaScript/Topics/Data_Structures/README.md:27]:
```suggestion
for (const x of [1,2,3,4,5]) {
```

**Review by Creatoon — CHANGES_REQUESTED** [review]

**Comment by daminimehra** [comment]
Done sir @creatoon :)

**Comment by paramsiddharth** [comment]
@Creatoon Please review further.

**Pull request merged**